Added a 3D geometry read from a .fits datacube

BertVdM opened the pull request:
**Description**
Added the functionality to read the spatial luminosity distribution for sources or the spatial density distribution for the medium from a .fits datacube.

**Motivation**
This allows SKIRT users to introduce arbitrary, asymmetric geometries into the simulation. 

**Tests**
This class has been thoroughly tested using a simple cubic geometry (cube.fits used by cube.ski). To exhibit the possibilities a top-toy was imported into SKIRT (top.fits used by top.ski). Test files are available here: [ReadFits3DGeometry_TestFiles.zip](https://github.com/SKIRT/SKIRT9/files/5495571/ReadFits3DGeometry_TestFiles.zip)


**Guidelines**
The model geometry is set by two parameters: the input filename and the pixelscale (i.e. the physical length per pixel).

**Context**
The example cube.fits represents a geometry of dimensions (7 AU x 4 AU x 10 AU), with pixelscale 1 AU (i.e. 7 x 4 x 10 spaxels).

petercamps commented:
Bert, good work. I pushed a new commit with streamlined formatting (using doxstyle for the comments and clang formatter for the code) and a few very minor changes to the code (e.g. renamed a temporary variable).

Before merging this pull request, I would like you to consider the following:
 - In the class header documentation, please include information on how to create the FITS file (i.e. a 3D data cube in the primary IFU; order of axis etc.). You could include an example of part of a FITS header to illustrate.
- Now that you removed the rotation, the `SigmaX,Y,Z()` functions could be reimplemented by simply adding the values of the spaxels along the appropriate axis. This would better match the intended purpose of these functions to represent the "theoretical" value of the input model.
- Can you add a test case with a truly asymmetrical 3D geometry (i.e. something more arbitrary than a cuboid).

BertVdM commented:
I have implemented these changes (probably without the appropriate formatting).
I have added a truly asymmetrical 3D geometry named "spiral.fits" to the test files:
[ReadFits3DGeometry_TestFilesUpdated.zip](https://github.com/SKIRT/SKIRT9/files/5502004/ReadFits3DGeometry_TestFilesUpdated.zip)

petercamps commented:
The density was not properly normalized with the cube of the pixel scale. When testing a geometry, you need to use it both as a source and a medium to ensure that all functions work. I also added a check for the maximum size of the data cube. Because we use int's as indices into the data cube, its total size cannot be larger than what fits in an int. I will merge the pull request as soon as the automated checks have completed.
